source/common/config/well_known_names.cc

```diff
@@ -0,0 +1,82 @@
+#include "common/config/well_known_names.h"
+
+namespace Envoy {
+namespace Config {
+
+std::unordered_map<std::string, std::string> TagNameValues::getRegexMapping() {
+  std::unordered_map<std::string, std::string> regex_mapping;
+
+  // cluster.<cluster_name>.
+  regex_mapping[CLUSTER_NAME] = "^cluster\\.(([^.]*)\\.)";
```

Comment:
Using the non-greedy operator here might be fine as well, I wasn't aware of it until you pointed it out. If you have sufficient unit tests for the tag extraction, it should be pretty clear how it's behaving. It's probably clearer to write as a non-greedy regex if it works in this scenario.